pkg/controllers/state/cluster.go

```diff
@@ -310,6 +320,10 @@ func (c *Cluster) UpdateNodeClaim(nodeClaim *v1.NodeClaim) {
 		n := c.newStateFromNodeClaim(nodeClaim, c.nodes[nodeClaim.Status.ProviderID])
 		c.nodes[nodeClaim.Status.ProviderID] = n
 	}
+
+	// Update nodepool state with NodeClaim
+	c.NodePoolState.UpdateNodeClaim(nodeClaim, c.IsNodeClaimMarkedForDeletion(nodeClaim.Status.ProviderID))
```

Comment:
Don't we already have a `n.MarkedForDeletion()` call? If we don't have a stateNode here yet, we can know off the bat that it's not marked for deletion yet

Reply:
Given that the calling function already checks if the providerID is set, could you consider moving this up to L321